First round of vectorizing "fixes"

jenshannoschwalm opened the pull request:
checked where simd omp might be wrong or missing

Fixes some clang reports about not able to vectorize, gcc didn't report

@ralfbrown would you have a look in detail and provide feedback? :-)

ralfbrown commented:
I'll take a look at the effects on GCC's code generation tomorrow, but other than the switch from MIN/MAX to fminf/fmaxf the changes seem OK.  Using fminf/fmaxf usually causes function calls with GCC.

jenshannoschwalm commented:
@ralfbrown , right. Fixed, squashed, added the rest of fixes - i think they are all good but keen to get your feedback.

There is one missing: src/common/imagebuf.c:163:1

ralfbrown commented:
> There is one missing: src/common/imagebuf.c:163:1

Looks to me like removing the "simd" from the OpenMP directive should fix that.  The only thing vectorizable in the body of that loop is inside its own OpenMP directive in the body of `copy_pixel_nontemporal`.

jenshannoschwalm commented:
Right :-)
